Update versions for latest .NET release

eerhardt opened the pull request:
9.0 RC2 shipped today. Along with a servicing release of .NET 8. Updating to these versions.
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/6194)

eerhardt committed:
Update versions for latest .NET release

joperezr commented on Directory.Packages.props:
```diff
@@ -195,7 +195,7 @@
 
     <!-- Pinned versions for Component Governance - Remove when root dependencies are updated -->
-    <PackageVersion Include="System.Text.Json" Version="8.0.4" />
+    <PackageVersion Include="System.Text.Json" Version="8.0.5" />
```
This was pinned for CG so I'm hesitant to keep it as 8.0.4. The intention would be that at some point, whoever is bringing that transitively would update themselves to something past 8.0.4 and we can remove the transitive pinning. If we keep bumping this, then that is unlikely to happen.

eerhardt replied:
8.0.4 is now marked vulnerable

https://www.nuget.org/packages/System.Text.Json/8.0.4

![image](https://github.com/user-attachments/assets/ab791b39-d9ef-4cfc-a6a6-20ca8a67d2dd)

joperezr replied:
gotcha, ok that totally makes sense then, sorry wasn't aware.

eerhardt committed:
Fix CPM dependencies on System.Format.Asn1 and System.Text.Json

When targeting net9.0, these need to be 9.0 dependencies, or else the EF packages complain that packages are being downgraded.

eerhardt commented:
/backport to release/9.0-rc1

sebastienros commented on eng/Versions.props:
```diff
@@ -39,56 +39,64 @@
+
+    <!-- System dependencies -->
+    <SystemFormatsAsn1PackageVersion>8.0.1</SystemFormatsAsn1PackageVersion>
+    <SystemTextJsonPackageVersion>8.0.5</SystemTextJsonPackageVersion>
```
Should we use the 9.0-rc one and this one for 8?

sebastienros replied:
Ok, it's below, sorry
